### DISCUSSION TITLE
Can i use Iperf to test 10G link

### DISCUSSION THREAD

**Discussion by Taitooooo** [esnet/iperf#1494] [Q&A]
Hello 
I need your inputs with commands if possible how can i use Iperf3 to test 10G link quality , i am targeting Error seconds ES , or PCS or EAS errors so i can test link quality 

Thanks

**Comment by bmah888** [comment]
Hi there! iperf3 works over pretty much any underlying link technology, in fact it's basically unaware of what links make up the IP routing path between the client and server. People have run iperf3 on everything from ISDN to 100G Ethernet. Although iperf3 doesn't directly measure the metrics you listed, you can definitely use it as a traffic generator, with other tools measuring the error rates, etc. Basically, set up two computers connected by the 10GE link you want to measure, and use iperf3 to send either TCP or UDP test traffic as appropriate. Any other specifics will depend on what exactly you want to measure, and under what circumstances.

**Reply by Taitooooo** [reply]
thanks for your answer , so if i installed it on server which is connecting to my network where we have many routers , we can use it as traffic generator and looking at client counters we can see errors
